src/Agent/NewRelic/Agent/Core/DistributedTracing/Samplers/AlwaysOffSampler.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+namespace NewRelic.Agent.Core.DistributedTracing.Samplers;
+
+public class AlwaysOffSampler: ISampler
+{
+    public ISamplingResult ShouldSample(ISamplingParameters samplingParameters) => new SamplingResult(false, 0.0f);
```

Comment:
We should be able to reduce allocations if we store this AlwaysOff sampling result as a static field in this class, and then this method can just return a reference to that static instance.